PYME/Analysis/tile_pyramid.py

```diff
@@ -254,7 +254,39 @@ def flush(self):
     def __del__(self):
         self._cur.close()
         self._conn.close()
-    
+
+
+TILEIO_EXT = {
+    '.pzf': PZFTileIO,
+    '.npy': NumpyTileIO,
+    '.db': SqliteTileIO,
+}
+
+def get_tileio_backend(base_directory):
+    """ find TileIO backend for a given ImagePyramid
+    
+    Parameters
+    ----------
+    base_directory : str
+        root directory of an ImagePyramid instance
+    
+    Returns
+    -------
+    class
+        which TileIO derived class the ImagePyramid can be
+        built with.
+    
+    Raises
+    ------
+    IOError
+        If no file with an extension in TILEIO_EXT is found.
+    """
+    for root, dirs, files in os.walk(base_directory):
+        for file in files:
+            file_extension = os.path.splitext(file)[-1]
+            if file_extension in TILEIO_EXT.keys():
+                return TILEIO_EXT[file_extension]
+    raise IOError("No files found for loading ImagePyramid.")
 
 class ImagePyramid(object):
     def __init__(self, storage_directory, pyramid_tile_size=256, mdh=None, 
```

Comment:
suggest adding `backend` parameter, defaulting to `backend=PZFTileIO` (i.e. the status quo). We need to specify this when creating a new pyramid, and the current default was chosen for performance - ie is probably the right one to use in most scenarios.